### ISSUE TITLE
Modularize itoolkit.js

### ISSUE THREAD

**Issue by abmusse** [IBM/nodejs-itoolkit#20]
I propose we modularize  the classes and methods within [itoolkit.js](https://github.com/IBM/nodejs-itoolkit/blob/master/lib/itoolkit.js)

  Move The following to there own files:

 - iConn class move to iconn.js
 - iPgm class mov to ipgm.js
 - iSql class move to isql.js
 - iCmd function move to icmd.js
 - iSh function move to ish.js
 - xmlToJson move to utils.js

For `xmlToJson` &  `__getClass` functions 

I propose we create a new file called `utils.js` and move both there since both are commonly used across the classes in the project.

Add exports to still allow access to all relevant Classes/Methods through itoolkit.js

In the end itoolkit.js will simply be our main entry point

```js
module.exports = {
  iConn: require('./iconn').iConn,
  iSh: require('./ish').iSh,
  iQsh: require('./iqsh').iQsh,
  iPgm: require('./ipgm').iPgm,
  iSql: require('./isql').iSql,
  iWork: require('./iwork').iWork,
  iProd: require('./iprod').iProd,
  iUserSpace: require('./iuserSpace').iUserSpace,
  iNetwork: require('./inetwork').iNetwork,
  iObj: require('./iobj').iObj,
  iDataQueue: require('./idataq').iDataQueue,
  xmlToJson: require('./utils').xmlToJson
};
```

@dmabupt @kadler @ThePrez @markdirish   

What do you all think?

If approved I will submit a PR with these changes

**Comment by markdirish** [comment]
I just looked at the file and with all the classes it looks like a good candidate for modularization. Just as a personal preference, when exporting a class from a file that only contains that class, I'm a fan of:

```
module.exports = class iConn {
...
}
```

And then in the itoolkit.js (or wherever) you only need ```const iConn = require(iConn);```, or as a property in this case. Also maybe call Object.freeze on module.exports just as a best practice so others can't change the properties.

I think your proposed changes will be good to start refactoring itoolkit.js, looks like it needs a little TLC.

**Comment by kadler** [comment]
SGTM.

**Comment by kadler** [comment]
> I just looked at the file and with all the classes it looks like a good candidate for modularization. Just as a personal preference, when exporting a class from a file that only contains that class, I'm a fan of:
> 
> ```
> module.exports = class iConn {
> ...
> }
> ```
> 
> And then in the itoolkit.js (or wherever) you only need `const iConn = require(iConn);`

This seems fine to me, so long as we don't expect the exports to ever change *or* the user is never expected to import iConn directly - just through itoolkit.

**Comment by abmusse** [comment]
Open PR #28 for this issue

@IBM/ibmi-open-source 

Feel free to review and comment

**Comment by abmusse** [comment]
PR merged in branch [v1.0-dev](https://github.com/IBM/nodejs-itoolkit/tree/v1.0-dev) closing this issue
